build_runner/lib/src/generate/build_impl.dart

```diff
@@ -92,7 +92,13 @@ Future<BuildResult> build(
   final buildPhases = await createBuildPhases(
       targetGraph, builders, builderConfigOverrides, isReleaseBuild ?? false);
 
-  var result = await singleBuild(environment, options, buildPhases);
+  BuildResult result;
+  if (buildPhases.isEmpty) {
+    _logger.severe('Nothing was built, yet a build was requested.');
```

Comment:
maybe "Nothing can be built, yet a build was requested."

We also want a similar check in watch mode around here: https://github.com/dart-lang/build/blob/0fdde9abb6bacbd8daf02806217278f27dd5b44d/build_runner/lib/src/generate/watch_impl.dart#L92

Reply:
Added the check. Should we exit as well? My initial thought is no.